tests/test_finite_difference.py

```diff
@@ -7,43 +7,37 @@ def setup_method():
     np.random.seed(0)
 
 
-@pytest.mark.parametrize(
-        'x_data', [np.arange(10), 3]
-)
-def test_model(x_data):
-    x, y, z = sf.variables('x, y, z')
-    a, b, c = sf.parameters('a, b, c')
-
-    '''Tests the case with 1 component and 1 parameter'''
-    model = sf.Model({y: 3 * a * x**2})
-    exact = model.eval_jacobian(x=x_data, a=3.5)
-    approx = model.finite_difference(x=x_data, a=3.5)
-    _assert_equal(exact, approx)
-
-    '''Tests the case with 1 component and multiple parameters'''
-    model = sf.Model({y: 3 * a * x**2 - sf.exp(b) * x})
-    exact = model.eval_jacobian(x=x_data, a=3.5, b=2)
-    approx = model.finite_difference(x=x_data, a=3.5, b=2)
-    _assert_equal(exact, approx)
+x, y, z = sf.variables('x, y, z')
+a, b, c = sf.parameters('a, b, c')
 
-    '''Tests the case with multiple components and one parameter'''
-    model = sf.Model({y: 3 * a * x**2,
-                      z: sf.exp(a*x)})
-    exact = model.eval_jacobian(x=x_data, a=3.5)
-    approx = model.finite_difference(x=x_data, a=3.5)
-    _assert_equal(exact, approx)
 
-    '''Tests the case with multiple components and multiple parameters'''
-    model = sf.Model({y: 3 * a * x**2 + b * x - c,
-                      z: sf.exp(a*x - b) * c})
-    exact = model.eval_jacobian(x=x_data, a=3.5, b=2, c=5)
-    approx = model.finite_difference(x=x_data, a=3.5, b=2, c=5)
-    _assert_equal(exact, approx, rel=1e-3)
+@pytest.mark.parametrize('x_data', [np.arange(10), 3])
+@pytest.mark.parametrize('_model,_init_par,relDiv',
+    [
+        ({y: 3 * a * x**2}, {'a': 3.5}, 1e-5),
+        ({y: 3 * a * x**2 - sf.exp(b) * x},
+        {'a': 3.5, 'b': 2}, 1e-5),
+        ({y: 3 * a * x**2, z: sf.exp(a*x)}, {'a': 3.5}, 1e-5),
+        ({y: 3 * a * x**2 + b * x - c, z: sf.exp(a*x - b) * c},
+            {'a': 3.5, 'b': 2, 'c': 5}, 1e-3)  
+    ]
+)
```

Comment:
Yes, awesome!
I have some minor linting issues (run pylint), but this is more or less what I meant.

Why do you prefix variables with _ here?
It is possible to choose points in parameter-space (abc) such that the reldiv can be the same for all?

Reply:
"minor issues"...alright :D Many of them will be away if we implement the collection of variables/parameters.

I used _ to highlight the model and parameters, but `init` is better. I changed the names of to `init_model `and `init_par` now. ;) 

Every test passed if we use `reldiv = 1e-5` as allowed difference.